sphinxext/opengraph/__init__.py

```diff
@@ -208,9 +262,14 @@ def setup(app: Sphinx) -> Dict[str, Any]:
     app.add_config_value("ogp_use_first_image", False, "html")
     app.add_config_value("ogp_type", "website", "html")
     app.add_config_value("ogp_site_name", None, "html")
+    app.add_config_value("ogp_social_cards", None, "html")
     app.add_config_value("ogp_custom_meta_tags", [], "html")
     app.add_config_value("ogp_enable_meta_description", True, "html")
 
+    # Social media card images
+    app.connect("builder-inited", setup_social_card_matplotlib_objects)
```

Comment:
I don't think there should be a separate setup function. If setup is needed, it would be better if the existing render function setup whatever it needed on its first call.